homework3/semi-uber/UI/src/app/regestration-page/regestration-page.component.ts

```diff
@@ -12,7 +15,11 @@ export class RegestrationPageComponent implements OnInit {
   passwordRegexp = "(?:[A-Z]+)(?:[a-z]+)(?:[0-9]+)(?:[@!#$%^&*()]+)";
   loginRegexp = ".{5,10}";
 
+  user: RegistrationModel;
+  formField = false;
+
   userData = new FormGroup({
+    name: new FormControl(""),
     username: new FormControl("", [
       Validators.required,
       Validators.pattern(this.loginRegexp)
```

Comment:
The are `maxLength` and `minLength` validators as well https://angular.io/api/forms/Validators.
So you do not need this pattern